### PR TITLE
Harden test for 'retrieve_metadata_server' against transparent DNS proxies

### PR DESCRIPTION
**Pull request by tseaver**
Grrr.

### CODE REVIEW

**Comment by tseaver** [comment]
The rationale:  I've seen the logging tests failing from my home machine for months now, due to that fact that my cable company intercepts DNS queries for unknown domains and redirects them to a "helpful" page.

Allowing a web request to propagate from inside a unit test is a Bad Thing(TM), in general.

File: logging/tests/unit/test__helpers.py [hunk 1]
```diff
@@ -105,15 +105,23 @@ def test_metadata_does_not_exist(self):
         self.assertIsNone(metadata)
 
     def test_request_exception(self):
+        import requests
+
         metadata_key = 'test_url_cannot_connect'
         metadata_url = 'http://metadata.invalid/'
 
-        patch = mock.patch(
+        requests_get_mock = mock.Mock()
```
Comment (dhermes) [inline logging/tests/unit/test__helpers.py]: Can you set the `spec`?

Comment (tseaver) [reply logging/tests/unit/test__helpers.py]: It is being used to mock a single function:  do we set `spec=['__call__']` as a rule in such cases?  I'd originally written it as an inline function:

```python
        def requests_get(*args, **kw):
              raise requests.exceptions.RequestsException('test')
```

which seemed clearer, but you have an allergy for them. :)

Comment (tseaver) [reply logging/tests/unit/test__helpers.py]: `git grep` showed a number of such cases, so I followed suit in 52c4b9f.

File: logging/tests/unit/test__helpers.py [hunk 2]
```diff
@@ -105,15 +105,23 @@ def test_metadata_does_not_exist(self):
         self.assertIsNone(metadata)
 
     def test_request_exception(self):
+        import requests
+
         metadata_key = 'test_url_cannot_connect'
         metadata_url = 'http://metadata.invalid/'
 
-        patch = mock.patch(
+        requests_get_mock = mock.Mock(spec=['__call__'])
```
Comment (dhermes) [inline logging/tests/unit/test__helpers.py]: @tseaver I think `spec=()` will produce the same results (i.e. `__call__` isn't actually necessary)

**Pull request merged**